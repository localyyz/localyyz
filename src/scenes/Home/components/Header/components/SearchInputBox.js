import React from "react";
import { View, StyleSheet, TextInput } from "react-native";
import { Styles, Sizes, Colours } from "localyyz/constants";

// third party
import { inject, observer } from "mobx-react/native";

@inject(stores => ({
  searchQuery: stores.homeStore.searchQuery,
  onChangeText: query => (stores.homeStore.searchQuery = query),

  // suggestions
  currentSuggestion:
    stores.homeStore.searchSuggestions[stores.homeStore.currentSuggestion]
}))
@observer
export default class SearchInputBox extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isEditing: false
    };

    // refs
    this.ref = React.createRef();

    // bindings
    this.onSubmitEditing = this.onSubmitEditing.bind(this);
    this.focus = this.focus.bind(this);
  }

  onSubmitEditing() {
    // use suggested search query since none was given
    if (!this.props.searchQuery) {
      this.props.onChangeText(this.props.currentSuggestion);
    }
  }

  focus() {
    this.ref.current && this.ref.current.focus();
  }

  render() {
    return (
      <View style={styles.container}>
        <TextInput
          autoFocus
          autoCorrect={false}
          ref={this.ref}
          returnKeyType="search"
          value={this.props.searchQuery}
          onChangeText={this.props.onChangeText}
          onSubmitEditing={this.onSubmitEditing}
          underlineColorAndroid={Colours.Transparent}
          style={styles.input}/>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    marginTop: 1,
    marginLeft: Sizes.InnerFrame * 3.5
  },

  input: {
    ...Styles.Text,
    ...Styles.Emphasized,
    ...Styles.Alternate,
    paddingRight: Sizes.InnerFrame * 2
  }
});
