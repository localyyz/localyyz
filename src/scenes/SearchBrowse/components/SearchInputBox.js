import React from "react";
import { View, StyleSheet, TextInput } from "react-native";

// third party
import { inject, observer } from "mobx-react/native";
import MaterialIcon from "react-native-vector-icons/MaterialIcons";

// custom
import { Styles, Sizes, Colours } from "localyyz/constants";

@inject(stores => ({
  searchQuery: stores.searchStore.searchQuery,
  setQuery: stores.searchStore.clearSearch,
  submit: stores.searchStore.reset,
  hasResults: stores.searchStore.hasResults,
  numProducts: stores.searchStore.numProducts
}))
@observer
export default class SearchInputBox extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      query: ""
    };

    // bindings
    this.onSubmitEditing = this.onSubmitEditing.bind(this);
    this.onChangeText = this.onChangeText.bind(this);
    this.clear = this.clear.bind(this);
  }

  onSubmitEditing() {
    // use suggested search query since none was given
    //
    // -> setting searchQuery is will trigger a mobx reaction
    // NOTE: see store for detail
    this.props.setQuery(this.state.query);

    // trigger the search
    this.props.submit();

    // callback from parent
    this.props.onSubmit && this.props.onSubmit();
  }

  onChangeText(text) {
    this.setState({
      query: text
    });
  }

  clear() {
    this.onChangeText("");
    this.props.setQuery();
  }

  render() {
    return (
      <View style={styles.container}>
        <MaterialIcon name="search" size={Sizes.Text} color={Colours.Text} />
        <TextInput
          autoCorrect={false}
          returnKeyType="search"
          value={this.state.query}
          onChangeText={this.onChangeText}
          onSubmitEditing={this.onSubmitEditing}
          underlineColorAndroid={Colours.Transparent}
          placeholder="Search over 200,000 products"
          style={styles.input}/>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    ...Styles.Horizontal,
    alignItems: "center",
    borderBottomWidth: Sizes.Spacer,
    borderBottomColor: Colours.SubduedText,
    paddingTop: Sizes.ScreenTop,
    paddingBottom: Sizes.InnerFrame / 2,
    marginLeft: Sizes.InnerFrame,
    marginBottom: Sizes.Height / 12,
    backgroundColor: Colours.Foreground
  },

  input: {
    ...Styles.Text,
    ...Styles.Emphasized,
    ...Styles.SmallText,
    flex: 1,
    paddingHorizontal: Sizes.InnerFrame / 2
  }
});
