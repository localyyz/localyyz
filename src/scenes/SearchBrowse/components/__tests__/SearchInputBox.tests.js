import React from "react";
import Renderer from "react-test-renderer";

import SearchInputBox from "../SearchInputBox";

describe("SearchInputBox", () => {
  it("should render properly", () => {
    let render = Renderer.create(<SearchInputBox.wrappedComponent />);
    expect(render.toJSON()).toMatchSnapshot();
  });
});
