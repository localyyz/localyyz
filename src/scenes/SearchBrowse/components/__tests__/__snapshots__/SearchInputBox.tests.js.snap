// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInputBox should render properly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderBottomColor": "#999999",
      "borderBottomWidth": 0.5,
      "flexDirection": "row",
      "justifyContent": "flex-start",
      "marginBottom": 111.16666666666667,
      "marginLeft": 15,
      "paddingBottom": 7.5,
      "paddingTop": 30,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "rgb(30, 30, 30)",
          "fontSize": 16,
        },
        undefined,
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    î¢¶
  </Text>
  <TextInput
    allowFontScaling={true}
    autoCorrect={false}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Search from over 2000+ products"
    returnKeyType="search"
    style={
      Object {
        "backgroundColor": "rgba(255, 255, 255, 0)",
        "color": "#000000",
        "flex": 1,
        "fontSize": 14,
        "fontWeight": "600",
        "paddingHorizontal": 7.5,
      }
    }
    underlineColorAndroid="rgba(255, 255, 255, 0)"
    value=""
  />
</View>
`;
